### PR TITLE
StateT of SeaTunnelSource should extend `Serializable`

### PR DESCRIPTION
**Pull request by lhyundeadsoul**
## Purpose of this pull request

1. StateT of SeaTunnelSource should extend `Serializable`,so that `org.apache.seatunnel.api.source.SeaTunnelSource.getEnumeratorStateSerializer` can support a default implementation without specify Generic Type.  In fact, StateT in all subclass of SeaTunnelSource extend `Serializable` already.
2.  remove repetitive dependency of fake in example module
3. postgre jar should be contained in container like mysql-java, so it should be  provided, not compile

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

File: seatunnel-api/src/main/java/org/apache/seatunnel/api/source/SeaTunnelSource.java [hunk 1]
```diff
@@ -94,6 +94,8 @@ SourceSplitEnumerator<SplitT, StateT> restoreEnumerator(SourceSplitEnumerator.Co
      *
      * @return enumerator state serializer.
      */
-    Serializer<StateT> getEnumeratorStateSerializer();
+    default Serializer<StateT> getEnumeratorStateSerializer(){
+        return new DefaultSerializer<>();
+    }
```
Comment (ashulin) [inline seatunnel-api/src/main/java/org/apache/seatunnel/api/source/SeaTunnelSource.java:99]:
```suggestion
    default Serializer<StateT> getEnumeratorStateSerializer() {
        return new DefaultSerializer<>();
    }
```
Keep the code style.

**Review by ashulin — COMMENTED** [review]
Could you please remove the code block in the implementation class?
```java
@Override
public Serializer<T> getEnumeratorStateSerializer() {
return new DefaultSerializer<>();
}
```

File: seatunnel-api/src/main/java/org/apache/seatunnel/api/source/SeaTunnelSource.java (hunk 1 above)
Comment (lhyundeadsoul) [reply seatunnel-api/src/main/java/org/apache/seatunnel/api/source/SeaTunnelSource.java:99]: done

**Comment by lhyundeadsoul** [comment]
@ashulin CI error :`Caused by: java.lang.ClassNotFoundException: org.apache.seatunnel.connectors.seatunnel.file.sink.local.LocalFileSink` has no matter with this PR. This error has already been in the current dev branch.

**Comment by ashulin** [comment]
@Hisoka-X PTAL.

**Review by Hisoka-X — APPROVED** [review]
LGTM

**Pull request merged**